doc/dev/schema.md

```diff
@@ -0,0 +1,174 @@
+call.) Each ``net.box`` connection has space schema for an instance it
+is connected to. Router can access space schema by using ``net.box``
+connection object contents.
+
+### When schema is used
+
+Space schema is used to flatten maps on `crud.*_object*` requests.
+Together with sharding info, the space schema is used to calculate
+``bucket_id`` to choose replicaset for a request.
+
+### How schema is reloaded
+
+Now there are no automatic reload of space schema supported by
+``net.box`` itself (see [tarantool/tarantool/6169](https://github.com/tarantool/tarantool/issues/6169)).
+``crud`` uses connection ``reload_schema`` handle (``ping`` wrapper, see
+[PR#111 comment](https://github.com/tarantool/crud/pull/111#issuecomment-765811556))
+to reload schema. ``crud`` reloads each replicaset schema. If there are
+several requests to reload info, the only one reload fiber is started
+and every request waits for its completion.
+
+### When schema is reloaded and operation is retried
+
+The basic logic is as follows: if something had failed and space schema
+mismatch could be the reason, reload the schema and retry. If it didn't
+help after ``N`` retries (now ``N`` is ``1``), pass the error to the
+user.
+
+Retry with reload is triggered if
+- space not found;
+- ``bucket_id`` calculation has failed due to any reason;
+- operation had failed on storage, hash check is on and hashes mismatch.
+
+Let's talk a bit more about the last one. To enable hash check, the user
+should pass `add_space_schema_hash` option to a request. This option is
+always enabled for `crud.*_object*` requests. If hashes mismatch, it
+means the router space schema is inconsistent with the storage space
+schema, so we reload it. For ``*_many`` methods, reload and retry
+happens only if all tuples had failed with hash mismatch; otherwise,
+errors are passed to a user.
+
+Retries are counted per function it wraps, so in fact there could
+be more reloads than ``N`` for a single request. For example, if a user
+calls `insert_object`, object flatten has failed (and has succeeded
+after reload) and then storage insert has failed and hashes mismatch,
+reload will happen twice and there will be two retries (though it's two
+different blocks of code will be retried independently). Or if a user
+calls `insert_object_many`, each tuple flatten is wrapped independently
+(though it is hard to imagine a real case of multiple flatten retries:
+user will need to pass a batch of tables with different structure and
+change the schema several times to match the current one; in reality it
+always will be success after a single retry or a fast fail).
+
+### Alternative approaches
+
+One of the alternatives considered was to ping a storage instance on
+each request to refresh schema (see [PR#111 comment](https://github.com/tarantool/crud/pull/111#discussion_r562757016)).
```

Comment:
It worth to briefly say, why an alternative was declined.

Reply:
Mentioned